hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -5893,7 +5893,7 @@ void reportBadBlocks(LocatedBlock[] blocks) throws IOException {
         String[] storageIDs = blocks[i].getStorageIDs();
         for (int j = 0; j < nodes.length; j++) {
           NameNode.stateChangeLog.info("*DIR* reportBadBlocks for block: {} on"
-              + " datanode: {}", blk, nodes[j].getXferAddr());
+              + " datanode: {}" + " client: {}", blk, nodes[j].getXferAddr(), Server.getRemoteIp());
```

Comment:
From my personal point of view, reportBadBlocks should be reported by DN. What does this have to do with the client?  `Server.getRemoteIp()` is somewhat expensive, is there a good enough reason for us to do this?

Reply:
Instead of ``Server.getRemoteIp()``, do  a ``getClientMachine()`` instead, it handles calls via RBF.
Fetching of this IP Address should be outside lock, doing this inside lock will hit performance. 
You are logging this inside the loop, so better extract a variable outside.

reportBadBlocks is there in both ClientProtocol as well as in DatanodeProtocol. Can check somewhat here for pointers
https://github.com/apache/hadoop/blob/ca3526da9283500643479e784a779fb7898b6627/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java#L1533-L1534

Reply:
> Instead of `Server.getRemoteIp()`, do a `getClientMachine()` instead, it handles calls via RBF. Fetching of this IP Address should be outside lock, doing this inside lock will hit performance. You are logging this inside the loop, so better extract a variable outside.
> 
> reportBadBlocks is there in both ClientProtocol as well as in DatanodeProtocol. Can check somewhat here for pointers
> 
> https://github.com/apache/hadoop/blob/ca3526da9283500643479e784a779fb7898b6627/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java#L1533-L1534

@ayushtkn @slfan1989  Thanks for reiew,  I moved getClientIp out of the for loop and write lock.
and getClientMachine is not accessible here, this function always invoked by an open file.